modules/platforms/dotnet/Apache.Ignite/Table/IPartition.cs

```diff
@@ -0,0 +1,28 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apache.Ignite.Table;
+
+using System.Diagnostics.CodeAnalysis;
+
+/// <summary>
+/// Table partition.
+/// </summary>
+[SuppressMessage("Design", "CA1040:Avoid empty interfaces", Justification = "Reviewed.")]
+public interface IPartition
+{
+}
```

Comment:
Hmm, should not we at least support comparison operations?

Reply:
Do you mean relative comparison, or just equality?

Value-based equality is supported by the `HashPartition` implementation (two instances with the same id are considered equal).

Reply:
After some thinking I agree. Added [IEquatable<T>](https://learn.microsoft.com/en-us/dotnet/api/system.iequatable-1?view=net-8.0) to the interface to make things clearer.